webapp/src/components/blockIconSelector.test.tsx

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ReactElement} from 'react'
+import {fireEvent, render, screen} from '@testing-library/react'
+
+import userEvent from '@testing-library/user-event'
+
+import '@testing-library/jest-dom'
+import {IntlProvider} from 'react-intl'
+
+import {BlockIcons} from '../blockIcons'
+import {TestBlockFactory} from '../test/testBlockFactory'
+
+import BlockIconSelector from './blockIconSelector'
+
+const wrapIntl = (children: ReactElement) => <IntlProvider locale='en'>{children}</IntlProvider>
+const board = TestBlockFactory.createBoard()
+const icon = BlockIcons.shared.randomIcon()
+
+jest.mock('../mutator', () => ({
+    changeIcon: jest.fn((id, oldIcon, newIcon) => {
+        board.fields.icon = newIcon
```

Comment:
@jespino @wiggin77 is this a "legit"/"good" way to mock the `mutator` for the unit test?

In the real app there happens a lot behind the scenes via the interaction with the server. But essentially it returns the same data that was sent and performs incremental update of the data model.

Is this approach treated like a contract that we can rely on in the unit tests replicating it via simple updates from mocks?

Reply:
Just one thing
I mocked the mutator: 
- to make sure I only test this component and not all of the services.
- to keep component actions in testing and get the correct test coverage.

Reply:
I reviewed mocked mutator:  I don't add implementation but now, I verify it been called

```
expect(mockedMutator.changeIcon).toBeCalledTimes(1)
expect(mockedMutator.changeIcon).toBeCalledWith(board.id, board.fields.icon, '👍')